Prod deploy failed on main

## Prod deploy failed

**Repo**: alexsiri7/reli
**SHA**: `22d947cb1fd9dd5822e5342ca5d5ddea48e6981d`
**Deploy run/status**: https://github.com/alexsiri7/reli/actions/runs/25227458546
**State**: failure
**Deployed at**: 2026-05-01T18:35:16Z

Auto-filed by `pipeline-health-cron.sh`. Prod deploy is a pipeline bottleneck — archon has been fired immediately. This issue is tagged `archon:in-progress` so the pickup cron will not double-fire.

### Steps
1. Inspect deploy logs at https://github.com/alexsiri7/reli/actions/runs/25227458546
2. Identify root cause (Railway config? env var? build error? migration?)
3. Fix on a branch, land, confirm next deploy goes green

## 🔍 Investigation: Prod deploy failed on `main` — RAILWAY_TOKEN expired (38th occurrence)

**Type**: `BUG` (infrastructure / secret rotation — agent-unactionable)

### Assessment

| Metric | Value | Reasoning |
|--------|-------|-----------|
| Severity | `HIGH` | Prod deploy pipeline fully blocked at the staging gate — no code can ship until the token is rotated; live app keeps serving traffic, no data loss, so not CRITICAL. |
| Complexity | `LOW` | Single GitHub Actions secret update by a human via railway.com — zero code changes; only "complexity" is the human handoff. |
| Confidence | `HIGH` | Workflow log shows the exact failure string `RAILWAY_TOKEN is invalid or expired: Not Authorized` from the `Validate Railway secrets` step — same `{me{id}}` probe that has fired 37 times before; `railway-token-health.yml` has failed 4 days running (2026-04-28 → 2026-05-01). |

---

### Problem Statement

Run [25227458546](https://github.com/alexsiri7/reli/actions/runs/25227458546) failed at `Validate Railway secrets` on commit `22d947c` (the merge SHA of PR #848 — the previous investigation). This is the **38th** RAILWAY_TOKEN expiration. The token lives in GitHub Actions secrets and **cannot be rotated by an agent** — railway.com requires human auth.

---

### Root Cause Analysis

WHY: Pipeline run `25227458546` ended in `failure`.
↓ BECAUSE: `Deploy to staging` exited 1 at the `Validate Railway secrets` step; downstream jobs `skipped`.
↓ BECAUSE: The token-probe `{me{id}}` call to `https://backboard.railway.app/graphql/v2` returned an auth error.
  Evidence: `2026-05-01T18:35:14.3742222Z ##[error]RAILWAY_TOKEN is invalid or expired: Not Authorized`
↓ ROOT CAUSE: The `RAILWAY_TOKEN` repository secret is no longer accepted by Railway's API.
  Evidence: `.github/workflows/staging-pipeline.yml:49-58` issues the probe; Railway returns `Not Authorized`. Independently, `railway-token-health.yml` runs `25049349913` (2026-04-28), `25105119767` (2026-04-29), `25161724763` (2026-04-30), `25211139148` (2026-05-01) all conclude `failure`.

---

### Implementation Plan

| Step | File | Change |
|------|------|--------|
| 1 | (GitHub secret `RAILWAY_TOKEN`) | **Human rotates** via railway.com → GitHub repo secrets (account/team-scoped token, not project token) |
| 2 | (none — codebase) | No code/workflow/runbook changes; agents cannot perform this action |
| 3 | (GitHub Actions UI) | `gh run rerun 25227458546 --failed` and confirm the pipeline reaches production |

<details>
<summary>📋 Detailed Implementation Steps</summary>

Per `CLAUDE.md` § "Railway Token Rotation":

> Agents cannot rotate the Railway API token. The token lives in GitHub Actions secrets (`RAILWAY_TOKEN`) and requires human access to railway.com.
> When CI fails with `RAILWAY_TOKEN is invalid or expired`:
> 1. Do NOT create a `.github/RAILWAY_TOKEN_ROTATION_*.md` file claiming rotation is done.
> 2. File a GitHub issue or send mail to mayor with the error details.
> 3. Direct the human to `docs/RAILWAY_TOKEN_ROTATION_742.md` for the rotation runbook.

**Human action — follow `docs/RAILWAY_TOKEN_ROTATION_742.md`:**

1. Log into railway.com.
2. Generate a new account/team-scoped API token (must satisfy `{me{id}}` — project tokens **cannot** answer that query, per web research).
3. Update GitHub Actions secret `RAILWAY_TOKEN` at https://github.com/alexsiri7/reli/settings/secrets/actions.
4. Re-run the failed pipeline: `gh run rerun 25227458546 --failed` (or push a no-op commit).
5. Confirm `Validate Railway secrets` passes and the deploy proceeds through `Deploy staging image to Railway` → `Wait for staging health` → `Staging E2E smoke tests` → `Deploy to production`.
6. Comment on this issue with the green run URL, remove `archon:in-progress`, close the issue.
7. Confirm the next scheduled `railway-token-health.yml` run goes green.

**Explicitly NOT done (per CLAUDE.md, Category 1 errors):**
- No `.github/RAILWAY_TOKEN_ROTATION_*.md` "rotation done" file.
- No edits to `.github/workflows/staging-pipeline.yml` — the validator is correctly designed; the failure mode is informative, not a bug.
- No edits to `docs/RAILWAY_TOKEN_ROTATION_742.md` (canonical runbook).

</details>

---

### Validation

```bash
gh run rerun 25227458546 --failed
gh run watch <new-run-id>
```

Pipeline must reach `Deploy to production` and complete `success`. Next scheduled `railway-token-health.yml` run must also conclude `success`.

---

### Next Step

Human rotates `RAILWAY_TOKEN` per `docs/RAILWAY_TOKEN_ROTATION_742.md`. No agent implementation is possible.

---
*Investigated by Claude • 2026-05-01T19:35:00Z*

## 🔍 Investigation: Prod deploy failed on `main` — RAILWAY_TOKEN expired (38th occurrence, 3rd pickup)

**Type**: `BUG` (infrastructure / secret rotation — agent-unactionable)

### Assessment

| Metric | Value | Reasoning |
|--------|-------|-----------|
| Severity | `HIGH` | Staging pipeline still blocked — three further runs (`25237747540`, `25229747557`, `25227442673`) failed at `Validate Railway secrets` since PR #851 merged; live app still serves traffic, so not CRITICAL. |
| Complexity | `LOW` | Single GitHub Actions secret update by a human via railway.com — zero code changes. |
| Confidence | `HIGH` | Latest failed run `25237747540` (2026-05-01T23:34:49Z) emits the exact string `RAILWAY_TOKEN is invalid or expired: Not Authorized` from `staging-pipeline.yml:55` — same signature as the 37 prior occurrences. |

---

### Problem Statement

Prior investigation (PR #851) merged 2026-05-01T19:30:09Z but **the secret has not been rotated** — staging-pipeline runs `25227458546` → `25229747557` → `25237747540` all subsequently failed with the same `Not Authorized` message. The pickup cron has re-queued #850 twice (2026-05-01T21:00 and 23:30) because no new PR was opened. Per `CLAUDE.md` § "Railway Token Rotation", **agents cannot rotate this token** — railway.com requires human auth.

---

### Root Cause Analysis

WHY: Staging pipeline run `25237747540` ended in `failure`.
↓ BECAUSE: `Deploy to staging` exited 1 at `Validate Railway secrets`.
↓ BECAUSE: The `{me{id}}` probe to `https://backboard.railway.app/graphql/v2` returned `Not Authorized`.
  Evidence: `2026-05-01T23:34:49.1344177Z ##[error]RAILWAY_TOKEN is invalid or expired: Not Authorized`
↓ ROOT CAUSE: `RAILWAY_TOKEN` is still rejected by Railway's API; not rotated since PR #851 merged.

---

### Implementation Plan

| Step | File | Change |
|------|------|--------|
| 1 | (GitHub secret `RAILWAY_TOKEN`) | **Human rotates** via railway.com → repo secrets (account/team-scoped, not project token) |
| 2 | (none — codebase) | No source/workflow/runbook changes; agents cannot perform this action |
| 3 | (GitHub Actions UI) | `gh run rerun 25237747540 --failed` and confirm pipeline reaches production |

<details>
<summary>📋 Detailed Implementation Steps</summary>

Per `CLAUDE.md` § "Railway Token Rotation":

> When CI fails with `RAILWAY_TOKEN is invalid or expired`:
> 1. Do NOT create a `.github/RAILWAY_TOKEN_ROTATION_*.md` file claiming rotation is done.
> 2. File a GitHub issue or send mail to mayor with the error details.
> 3. Direct the human to `docs/RAILWAY_TOKEN_ROTATION_742.md` for the rotation runbook.

**Human action — follow `docs/RAILWAY_TOKEN_ROTATION_742.md`:**

1. Log into railway.com.
2. Generate a new **account/team-scoped** API token at `https://railway.com/account/tokens` (workspace and project tokens cannot answer `{me{id}}`).
3. Update GitHub Actions secret `RAILWAY_TOKEN` at https://github.com/alexsiri7/reli/settings/secrets/actions.
4. Re-run failed pipeline: `gh run rerun 25237747540 --failed`.
5. Confirm `Validate Railway secrets` passes and the deploy proceeds through `Deploy to production`.
6. Comment here with the green run URL, remove `archon:in-progress`, close this issue.
7. Confirm next scheduled `railway-token-health.yml` run also goes green.

**Explicitly NOT done (Category 1 traps per `CLAUDE.md`):**
- No `.github/RAILWAY_TOKEN_ROTATION_*.md` "rotation done" file.
- No edits to `.github/workflows/staging-pipeline.yml` — the validator is correctly designed.
- No edits to `docs/RAILWAY_TOKEN_ROTATION_742.md` — canonical runbook owned by separate change.
- No swap to `Project-Access-Token` headers — project tokens cannot call `serviceInstanceUpdate`/`serviceInstanceDeploy` (per `web-research.md` from PR #851).

</details>

---

### Validation

```bash
gh run rerun 25237747540 --failed
gh run list --workflow=staging-pipeline.yml --limit 1
gh run list --workflow=railway-token-health.yml --limit 1
```

Both workflows must show `success` after rotation.

---

### Next Step

Human rotates `RAILWAY_TOKEN` per `docs/RAILWAY_TOKEN_ROTATION_742.md`. No agent implementation is possible — this pickup adds an investigation artifact only.

---
*Investigated by Claude • 2026-05-02T00:00:00Z • workflow `143fba6af8ed9a6eee7eae7c6cc02a7d`*

## 🔍 Investigation: Prod deploy failed on `main` — RAILWAY_TOKEN expired (38th occurrence, 5th pickup)

**Type**: `BUG` (infrastructure / secret rotation — agent-unactionable)

### Assessment

| Metric | Value | Reasoning |
|--------|-------|-----------|
| Severity | `HIGH` | Staging→production pipeline fully blocked at `Validate Railway secrets`; no code can ship until rotation. Live app still serves traffic, no data loss → not CRITICAL. |
| Complexity | `LOW` | Single GitHub Actions secret update by a human via railway.com — zero code/workflow/runbook changes. |
| Confidence | `HIGH` | Latest run [`25239867327`](https://github.com/alexsiri7/reli/actions/runs/25239867327) (2026-05-02T01:04:55Z) emits the exact string `RAILWAY_TOKEN is invalid or expired: Not Authorized` from `staging-pipeline.yml:55` — identical to the 37 prior occurrences. Five consecutive pipeline runs and four consecutive scheduled `railway-token-health.yml` runs (2026-04-28 → 2026-05-01) all fail at the same probe. |

---

### Problem Statement

The staging pipeline has now failed five times in a row (`25227458546` → `25229747557` → `25237747540` → `25238494833` → `25239867327`) at `Validate Railway secrets`. Prior pickups (PRs #851, #853) landed investigation artifacts only; the secret has not been rotated. Issue #854 is also open for the 39th occurrence (PR #855 merged the artifact). The pickup cron has now re-queued #850 four times because **agents cannot rotate this token** — `CLAUDE.md` § "Railway Token Rotation" requires a human with railway.com access.

---

### Root Cause Analysis

WHY: Staging pipeline run [`25239867327`](https://github.com/alexsiri7/reli/actions/runs/25239867327) ended in `failure`.
↓ BECAUSE: `Deploy to staging` exited 1 at `Validate Railway secrets`; downstream jobs skipped.
↓ BECAUSE: The `{me{id}}` probe to `https://backboard.railway.app/graphql/v2` returned an auth error.
  Evidence: `2026-05-02T01:04:55.5764191Z ##[error]RAILWAY_TOKEN is invalid or expired: Not Authorized`
↓ ROOT CAUSE: The `RAILWAY_TOKEN` repository secret is not accepted by Railway's API; not rotated since PR #853 merged.

---

### Implementation Plan

| Step | File | Change |
|------|------|--------|
| 1 | (GitHub secret `RAILWAY_TOKEN`) | **Human rotates** via railway.com → repo Actions secrets (account/team-scoped, NOT a project token) |
| 2 | (none — codebase) | No source / workflow / runbook changes; agents cannot perform this action |
| 3 | (GitHub Actions UI) | `gh run rerun 25239867327 --failed` and confirm pipeline reaches `Deploy to production` |

<details>
<summary>📋 Detailed Implementation Steps</summary>

Per `CLAUDE.md` § "Railway Token Rotation":

> Agents cannot rotate the Railway API token. The token lives in GitHub Actions secrets (`RAILWAY_TOKEN`) and requires human access to railway.com.
> When CI fails with `RAILWAY_TOKEN is invalid or expired`:
> 1. Do NOT create a `.github/RAILWAY_TOKEN_ROTATION_*.md` file claiming rotation is done.
> 2. File a GitHub issue or send mail to mayor with the error details.
> 3. Direct the human to `docs/RAILWAY_TOKEN_ROTATION_742.md` for the rotation runbook.

**Human action — follow `docs/RAILWAY_TOKEN_ROTATION_742.md`:**

1. Log into railway.com.
2. Generate a new **account/team-scoped** API token at https://railway.com/account/tokens (project tokens cannot satisfy `{me{id}}` — see web-research.md from PR #853).
3. Update GitHub Actions secret `RAILWAY_TOKEN` at https://github.com/alexsiri7/reli/settings/secrets/actions.
4. Re-run failed pipeline: `gh run rerun 25239867327 --failed`.
5. Confirm `Validate Railway secrets` passes and the deploy proceeds through `Deploy to production`.
6. Comment here with the green run URL, remove `archon:in-progress`, close #850 (and #854).
7. Confirm the next scheduled `railway-token-health.yml` run also goes green.

**Explicitly NOT done (Category 1 traps per `CLAUDE.md`):**
- No `.github/RAILWAY_TOKEN_ROTATION_*.md` "rotation done" file.
- No edits to `.github/workflows/staging-pipeline.yml` — validator is correctly designed.
- No edits to `.github/workflows/railway-token-health.yml` — probe is correctly designed.
- No edits to `docs/RAILWAY_TOKEN_ROTATION_742.md` — canonical runbook, owned by separate change.
- No swap to `Project-Access-Token` headers — project tokens cannot call `serviceInstanceUpdate`/`serviceInstanceDeploy` (per web-research.md from PR #853).

</details>

---

### Validation

```bash
gh run rerun 25239867327 --failed
gh run list --workflow=staging-pipeline.yml --limit 1
gh run list --workflow=railway-token-health.yml --limit 1
```

Both workflows must show `success` after rotation.

---

### Next Step

Human rotates `RAILWAY_TOKEN` per [`docs/RAILWAY_TOKEN_ROTATION_742.md`](https://github.com/alexsiri7/reli/blob/main/docs/RAILWAY_TOKEN_ROTATION_742.md). No agent implementation is possible — this pickup adds an investigation artifact only.

Until the human rotation lands, the pickup cron will continue to re-queue this issue.

---
*Investigated by Claude • 2026-05-02T02:15:00Z • workflow `ada4a84b65f08b01b649caa2de5524dc`*

--- a/artifacts/runs/ada4a84b65f08b01b649caa2de5524dc/investigation.md
+++ b/artifacts/runs/ada4a84b65f08b01b649caa2de5524dc/investigation.md
@@ -89,7 +89,7 @@ Per `CLAUDE.md` § "Railway Token Rotation":
 The issue is already open (#850); this artifact is the agent's contribution. The human steps are:
 
 1. Log into railway.com.
-2. Generate a new **account/team-scoped** API token at https://railway.com/account/tokens (project tokens **cannot** answer the `{me{id}}` probe — see web-research.md §2).
+2. Generate a new **account/team-scoped** API token at https://railway.com/account/tokens. The validator at `staging-pipeline.yml:49-58` issues `Authorization: Bearer` + `{me{id}}`, which a project token cannot satisfy (see web-research.md §3). If the rotated token still fails, capture `errors[0].message` from the failure log and consult web-research.md §2 for the contradicting community report on token-type acceptance.
 3. Update GitHub Actions secret `RAILWAY_TOKEN` at https://github.com/alexsiri7/reli/settings/secrets/actions.
 4. Re-run failed pipeline: `gh run rerun 25239867327 --failed`.
 5. Confirm `Validate Railway secrets` passes and the deploy proceeds through `Deploy staging image to Railway` → `Wait for staging health` → `Staging E2E smoke tests` → `Deploy to production`.
@@ -97,6 +97,8 @@ The issue is already open (#850); this artifact is the agent's contribution. The
 7. Confirm the next scheduled `railway-token-health.yml` run goes green.
 
 **Why**: The validator at `staging-pipeline.yml:53-57` correctly identifies that `{me{id}}` returns no `data.me.id` — the only way to make this pass is to supply a token Railway accepts.
+
+> **Runbook drift note (out of scope for this PR; flag to mayor):** The canonical runbook `docs/RAILWAY_TOKEN_ROTATION_742.md:24-26` is silent on token type (account vs. workspace vs. project). This investigation tightens that to "account/team-scoped, NOT a project token" based on the validator's `{me{id}}` probe. Per CLAUDE.md § "Polecat Scope Discipline", the runbook itself is owned by a separate change — flagging here so a follow-up bead can fold the token-type constraint into the runbook (may explain part of the 38-occurrence cadence if prior rotations used the wrong type).
 
 ### Step 2: No code changes
 

--- a/artifacts/runs/ada4a84b65f08b01b649caa2de5524dc/investigation.md
+++ b/artifacts/runs/ada4a84b65f08b01b649caa2de5524dc/investigation.md
@@ -1,0 +1,189 @@
+# Investigation: Prod deploy failed on `main` — RAILWAY_TOKEN expired (38th occurrence, 5th pickup)
+
+**Issue**: #850 (https://github.com/alexsiri7/reli/issues/850)
+**Type**: BUG (infrastructure / secret rotation — agent-unactionable)
+**Investigated**: 2026-05-02T02:15:00Z
+**Workflow**: ada4a84b65f08b01b649caa2de5524dc
+
+### Assessment
+
+| Metric | Value | Reasoning |
+|--------|-------|-----------|
+| Severity | HIGH | Staging→production pipeline fully blocked at `Validate Railway secrets`; no code can ship until the token is rotated. Live app keeps serving traffic and there is no data loss, so not CRITICAL. |
+| Complexity | LOW | Fix is a single GitHub Actions secret update by a human via railway.com — zero code, workflow, or runbook changes are needed; only complexity is the human handoff. |
+| Confidence | HIGH | Latest failed run `25239867327` (2026-05-02T01:04:55Z) emits the exact string `RAILWAY_TOKEN is invalid or expired: Not Authorized` from `staging-pipeline.yml:55`, identical to the 37 prior occurrences. Five consecutive `staging-pipeline` runs (`25227458546`, `25229747557`, `25237747540`, `25238494833`, `25239867327`) and four consecutive scheduled `railway-token-health.yml` runs (2026-04-28 → 2026-05-01) all fail at the same probe. |
+
+---
+
+## Problem Statement
+
+The staging pipeline has now failed five times in a row (`25227458546` → `25239867327`) at the `Validate Railway secrets` step with `Not Authorized` from Railway's `{me{id}}` probe. PRs #851 and #853 landed investigation artifacts for this same issue (#850); they did not — and could not — rotate the underlying secret. Issue #854 already opened for the 39th occurrence (PR #855 merged the artifact). The pickup cron has now re-queued #850 four times because no PR followed each prior pickup with the actual fix. **Per `CLAUDE.md` § "Railway Token Rotation", agents cannot perform this rotation** — the fix lives in a non-code primitive (a GitHub Actions secret) that requires human auth at railway.com.
+
+---
+
+## Analysis
+
+### First-Principles
+
+| Primitive | File:Lines | Sound? | Notes |
+|-----------|-----------|--------|-------|
+| `Validate Railway secrets` probe | `.github/workflows/staging-pipeline.yml:49-58` | Yes | Correctly fails fast with an actionable error message. The validator is doing exactly what it should — surfacing the expired secret before a destructive deploy attempt. |
+| `RAILWAY_TOKEN` GitHub Actions secret | (GitHub repo settings → Actions secrets) | No (operational) | Token is invalid; secret needs to be re-issued by a human via railway.com. Agents have no credential to authenticate to railway.com. |
+| Rotation runbook | `docs/RAILWAY_TOKEN_ROTATION_742.md` | Yes | Canonical, human-targeted runbook already exists. No edits needed. |
+
+The **primitive is sound**; the **operational state is broken**. There is no code change that can fix this — only a human secret rotation.
+
+### Root Cause / Evidence Chain
+
+WHY: Staging pipeline run `25239867327` ended in `failure`.
+↓ BECAUSE: Job `Deploy to staging` exited 1 at the `Validate Railway secrets` step; downstream jobs (`Deploy staging image to Railway`, `Wait for staging health`, `Staging E2E smoke tests`, `Deploy to production`) were skipped.
+↓ BECAUSE: The `{me{id}}` probe (`.github/workflows/staging-pipeline.yml:49-52`) to `https://backboard.railway.app/graphql/v2` returned an auth error.
+  Evidence: `2026-05-02T01:04:55.5764191Z ##[error]RAILWAY_TOKEN is invalid or expired: Not Authorized`
+↓ ROOT CAUSE: The `RAILWAY_TOKEN` repository secret is not accepted by Railway's API.
+  Evidence: same `Not Authorized` signature across `25227458546` (18:35), `25229747557` (19:34), `25237747540` (23:34), `25238494833` (00:04), `25239867327` (01:04). Independently, the scheduled `railway-token-health.yml` job has failed every day from 2026-04-28 through 2026-05-01 (`25049349913`, `25105119767`, `25161724763`, `25211139148`).
+
+### Affected Files
+
+| File | Lines | Action | Description |
+|------|-------|--------|-------------|
+| (GitHub Actions secret `RAILWAY_TOKEN`) | n/a | UPDATE (human) | Replace with a freshly issued account/team-scoped Railway API token |
+| `.github/workflows/staging-pipeline.yml` | 49-58 | (no change) | Validator works as designed — confirmed by clean `Not Authorized` surfacing |
+| `docs/RAILWAY_TOKEN_ROTATION_742.md` | n/a | (no change) | Canonical runbook — owned by separate change, do not edit here |
+| `.github/RAILWAY_TOKEN_ROTATION_*.md` | n/a | **DO NOT CREATE** | Category 1 trap per `CLAUDE.md` |
+
+### Integration Points
+
+- `.github/workflows/staging-pipeline.yml` consumes `RAILWAY_TOKEN` at `Validate Railway secrets` (line 50), `Deploy staging image to Railway` (line 62), and downstream Railway GraphQL calls.
+- `.github/workflows/railway-token-health.yml` (scheduled) probes the same secret daily — currently red since 2026-04-28.
+- No code paths consume `RAILWAY_TOKEN` (it is a CI-only secret).
+
+### Git History
+
+- `staging-pipeline.yml` validator step has been stable for many weeks; this is not a regression in the workflow.
+- 38 prior recurrences across issues (e.g. `docs/RAILWAY_TOKEN_ROTATION_742.md` was created for the 1st rotation; subsequent investigations have followed the same pattern: `#843`, `#847`, `#850`, `#854`).
+- **Implication**: This is a **recurring operational issue**, not a code defect. The high cadence (38 occurrences) is itself a signal worth investigating in a separate bead — see web-research.md for hypotheses (wrong token type stored in the secret, account-side revocation, leakage triggering invalidation).
+
+---
+
+## Implementation Plan
+
+| Step | File | Change |
+|------|------|--------|
+| 1 | (GitHub secret `RAILWAY_TOKEN`) | **Human rotates** via railway.com → repo Actions secrets (account/team-scoped token, NOT a project token) |
+| 2 | (none — codebase) | No source / workflow / runbook changes; agents cannot perform this action |
+| 3 | (GitHub Actions UI / `gh` CLI) | `gh run rerun 25239867327 --failed` and confirm pipeline reaches `Deploy to production` |
+
+### Step 1: Human rotates `RAILWAY_TOKEN` per the canonical runbook
+
+**File**: GitHub Actions secret (no source file)
+**Action**: UPDATE (human-only)
+
+Per `CLAUDE.md` § "Railway Token Rotation":
+
+> Agents cannot rotate the Railway API token. The token lives in GitHub Actions secrets (`RAILWAY_TOKEN`) and requires human access to railway.com.
+> When CI fails with `RAILWAY_TOKEN is invalid or expired`:
+> 1. Do NOT create a `.github/RAILWAY_TOKEN_ROTATION_*.md` file claiming rotation is done.
+> 2. File a GitHub issue or send mail to mayor with the error details.
+> 3. Direct the human to `docs/RAILWAY_TOKEN_ROTATION_742.md` for the rotation runbook.
+
+The issue is already open (#850); this artifact is the agent's contribution. The human steps are:
+
+1. Log into railway.com.
+2. Generate a new **account/team-scoped** API token at https://railway.com/account/tokens (project tokens **cannot** answer the `{me{id}}` probe — see web-research.md §2).
+3. Update GitHub Actions secret `RAILWAY_TOKEN` at https://github.com/alexsiri7/reli/settings/secrets/actions.
+4. Re-run failed pipeline: `gh run rerun 25239867327 --failed`.
+5. Confirm `Validate Railway secrets` passes and the deploy proceeds through `Deploy staging image to Railway` → `Wait for staging health` → `Staging E2E smoke tests` → `Deploy to production`.
+6. Comment on this issue with the green run URL, remove `archon:in-progress`, close #850 (and #854).
+7. Confirm the next scheduled `railway-token-health.yml` run goes green.
+
+**Why**: The validator at `staging-pipeline.yml:53-57` correctly identifies that `{me{id}}` returns no `data.me.id` — the only way to make this pass is to supply a token Railway accepts.
+
+### Step 2: No code changes
+
+The validator, deploy step, and runbook are all correct. Editing any of them would either (a) create a Category 1 documentation trap (claiming work an agent cannot do) or (b) destabilize a known-working primitive without addressing the root cause.
+
+### Step 3: Re-run pipeline post-rotation
+
+`gh run rerun 25239867327 --failed` after the secret is updated. Verify via `gh run watch <new-run-id>` that the deploy reaches `Deploy to production`.
+
+---
+
+## Patterns to Follow
+
+This is the **5th pickup** of issue #850. The established pattern across the prior 4 pickups (PRs #851, #853, plus two re-queues) is:
+
+```
+# SOURCE: pattern used by PR #851 and PR #853
+# 1. Write investigation artifact under artifacts/runs/<workflow-id>/
+# 2. Reference latest failed run ID and exact error string
+# 3. Cite CLAUDE.md § "Railway Token Rotation" verbatim
+# 4. Direct human to docs/RAILWAY_TOKEN_ROTATION_742.md
+# 5. Explicitly enumerate the Category 1 traps NOT taken
+# 6. Post a GH comment summarising the artifact
+# 7. Make NO source / workflow / runbook edits
+```
+
+This pickup follows that pattern exactly.
+
+---
+
+## Edge Cases & Risks
+
+| Risk / Edge Case | Mitigation |
+|------------------|------------|
+| Agent invents a "rotation done" `.github/RAILWAY_TOKEN_ROTATION_*.md` | Explicitly enumerated as not-done in this artifact and in the GH comment; CLAUDE.md flags this as a Category 1 error |
+| Agent edits `staging-pipeline.yml` to bypass the validator | Validator is correct and load-bearing; bypassing it would let a broken deploy proceed and damage prod |
+| Agent edits the canonical runbook `docs/RAILWAY_TOKEN_ROTATION_742.md` | Out of scope; runbook is owned by a separate change. Edits here would be an out-of-scope drive-by |
+| Agent attempts to swap to `Project-Access-Token` header | Per web-research.md from PR #851: project tokens **cannot** call `serviceInstanceUpdate`/`serviceInstanceDeploy`, so this would break the deploy step too |
+| Pickup cron continues to re-queue indefinitely | This is by design; the cron unblocks itself only when a human rotates the secret. The re-queue cost is bounded (one investigation artifact per pickup) |
+
+---
+
+## Validation
+
+### Automated Checks
+
+```bash
+# After human rotation:
+gh run rerun 25239867327 --failed
+gh run list --workflow=staging-pipeline.yml --limit 1
+gh run list --workflow=railway-token-health.yml --limit 1
+```
+
+Both workflows must show `success` after rotation.
+
+### Manual Verification
+
+1. Pipeline reaches `Deploy to production` (not skipped).
+2. Production health check passes.
+3. Next scheduled `railway-token-health.yml` run goes green.
+4. Issues #850 and #854 are closed by the human operator.
+
+---
+
+## Scope Boundaries
+
+**IN SCOPE:**
+- Write this investigation artifact.
+- Post a summary comment on issue #850.
+- Direct the human to the canonical rotation runbook.
+
+**OUT OF SCOPE (do not touch):**
+- `.github/workflows/staging-pipeline.yml` (validator is correct).
+- `.github/workflows/railway-token-health.yml` (probe is correct).
+- `docs/RAILWAY_TOKEN_ROTATION_742.md` (canonical runbook, separate change).
+- Any `.github/RAILWAY_TOKEN_ROTATION_*.md` file (Category 1 trap).
+- Renaming `RAILWAY_TOKEN` → `RAILWAY_API_TOKEN` (deferred — see PR #853 follow-ups).
+- Refactor to Railway CLI + project token (deferred — see PR #853 follow-ups, ~half-day durable refactor).
+- Closing or deduping issue #854 (separate triage decision for the human operator).
+
+---
+
+## Metadata
+
+- **Investigated by**: Claude (Opus 4.7)
+- **Timestamp**: 2026-05-02T02:15:00Z
+- **Workflow ID**: ada4a84b65f08b01b649caa2de5524dc
+- **Artifact**: `artifacts/runs/ada4a84b65f08b01b649caa2de5524dc/investigation.md`
+- **Companion**: `artifacts/runs/ada4a84b65f08b01b649caa2de5524dc/web-research.md`
+- **Pickup count**: 5th pickup of #850 (initial PR #851 → re-queue → re-queue → PR #853 → re-queue at 2026-05-02T02:00:42Z → this pickup)

--- a/artifacts/runs/ada4a84b65f08b01b649caa2de5524dc/web-research.md
+++ b/artifacts/runs/ada4a84b65f08b01b649caa2de5524dc/web-research.md
@@ -1,0 +1,211 @@
+# Web Research: fix #850
+
+**Researched**: 2026-05-02T00:00:00Z
+**Workflow ID**: ada4a84b65f08b01b649caa2de5524dc
+**Issue**: #850 — Prod deploy failed on main (38th `RAILWAY_TOKEN` expiration; 4 prior pickups on this issue alone)
+
+---
+
+## Summary
+
+Issue #850 is the 38th recurrence of the `RAILWAY_TOKEN is invalid or expired: Not Authorized` failure in `.github/workflows/staging-pipeline.yml`. Research shows two likely durable-fix paths: (1) ensure the secret holds a **project-scoped token** (created in Project → Settings → Tokens), because Railway's `RAILWAY_TOKEN` env var rejects account/workspace tokens with the exact "invalid or expired" message even immediately after creation; and (2) Railway does not document OIDC federation, so the long-lived secret cannot be eliminated — but project tokens, when correctly scoped, do not appear to have a documented automatic expiration. The 38-failure cadence strongly suggests something other than natural TTL is at play (wrong token type, account-level revocation, or token leakage triggering automatic invalidation).
+
+**Per CLAUDE.md, this agent cannot rotate the token.** The artifact below informs the human operator's next rotation and proposes structural fixes to break the recurrence cycle.
+
+---
+
+## Findings
+
+### 1. Railway token types and which one `RAILWAY_TOKEN` accepts
+
+**Source**: [Railway Public API Documentation](https://docs.railway.com/integrations/api)
+**Authority**: Official Railway documentation
+**Relevant to**: Root-cause hypothesis for the recurrence pattern
+
+**Key Information**:
+
+- Railway exposes four token types: **Account Token** (all resources/workspaces), **Workspace Token** (single workspace, recommended for "Team CI/CD, shared automation"), **Project Token** (single environment in a project, recommended for "Deployments, service-specific automation"), and **OAuth**.
+- Project tokens use a different HTTP header — `Project-Access-Token: <TOKEN>` — instead of `Authorization: Bearer <TOKEN>`.
+- Account/workspace tokens are created from the **account settings** tokens page; project tokens are created from the **project settings** tokens page.
+- The public docs do **not** document expiration/TTL for account, workspace, or project tokens. (TTL is only documented for OAuth: access tokens 1 hour, refresh tokens 1 year.)
+
+---
+
+### 2. `RAILWAY_TOKEN` only accepts project tokens — account tokens always show "invalid or expired"
+
+**Source**: [Railway Help Station — "RAILWAY_TOKEN invalid or expired"](https://station.railway.com/questions/railway-token-invalid-or-expired-59011e20)
+**Authority**: Community Q&A with confirmed working resolution
+**Relevant to**: Most likely root cause — wrong token type, not actual expiration
+
+**Key Information** (direct quote):
+
+> "RAILWAY_TOKEN now only accepts *project token*, if u put the normal account token (the one u make in account settings) it literally says 'invalid or expired' even if u just made it 2 seconds ago."
+
+**Recommended fix from the same thread**:
+
+> "go to your PROJECT (not account page) → settings → tokens → generate new project token there copy that shit put it as RAILWAY_TOKEN"
+
+Additional advice: remove or rename any `RAILWAY_API_TOKEN` env var if present, as it can interfere.
+
+> If both environment variables are set, `RAILWAY_TOKEN` takes precedence.
+> — [Railway Docs — Using the CLI](https://docs.railway.com/guides/cli)
+
+---
+
+### 3. The repository's deploy workflow validates the token via the GraphQL API directly
+
+**Source**: Local file `.github/workflows/staging-pipeline.yml` (per failed-run log of run 25227458546)
+**Relevant to**: The validation method may itself reject project tokens
+
+**Key Information**:
+
+The "Validate Railway secrets" step in the failing run does:
+
+```bash
+curl -sf -X POST "https://backboard.railway.app/graphql/v2" \
+  -H "Authorization: ***" \
+  -H "Content-Type: application/json" \
+  -d '{"query":"{me{id}}"}'
+```
+
+This uses `Authorization: Bearer <token>` and queries `{me{id}}`. Per the API docs (Finding #1), **project tokens use the `Project-Access-Token` header, not `Authorization: Bearer`**, and `me` is an account-scoped query that a project token may not be allowed to resolve. So even a freshly minted project token could fail this validation.
+
+This means the existing validation step may force the use of an account/workspace token — but Railway's CLI-side `RAILWAY_TOKEN` only accepts project tokens (Finding #2). The validation step and the deploy step may require **different** token types.
+
+---
+
+### 4. Railway's official recommendation for GitHub Actions
+
+**Source**: [Railway Blog — Using GitHub Actions with Railway](https://blog.railway.com/p/github-actions)
+**Authority**: Railway official blog
+**Relevant to**: The supported, intended deploy pattern
+
+**Key Information**:
+
+- Railway recommends **project tokens** for GitHub Actions deployment.
+- Project tokens are created in **project dashboard → Settings → Tokens** ("Project tokens allow the CLI to access all the environment variables associated with a specific project and environment").
+- Example workflow uses the Railway CLI Docker image, stores `RAILWAY_TOKEN` as a repo secret, and runs `railway up --service=${{ env.SVC_ID }}`.
+- The token "must be kept secret"; the service ID can be public.
+
+> "If you are using a team project, you need to ensure that the token specified is scoped to your account, not a workspace."
+> — Railway moderator, [Token for GitHub Action thread](https://station.railway.com/questions/token-for-git-hub-action-53342720)
+
+---
+
+### 5. Other known failure modes
+
+**Source**: [Railway Help Station — CLI throwing Unauthorized with RAILWAY_TOKEN](https://station.railway.com/questions/cli-throwing-unauthorized-with-railway-24883ba1)
+**Relevant to**: Alternative root causes besides expiration
+
+**Key Information**:
+
+- Local config corruption (`~/.railway/config.json`) can produce identical-looking unauthorized errors. Not relevant for GitHub-Actions-hosted runners (no persistent home).
+- Different commands require different scopes: `railway whoami` and `railway link` need an account-scoped token; `railway up` works with a project token.
+- A historical CLI bug (now fixed) caused `RAILWAY_API_TOKEN` to not be recognized properly.
+
+---
+
+### 6. Modern best practice: OIDC federation (not currently supported by Railway)
+
+**Source**: [GitHub OpenID Connect Documentation](https://docs.github.com/en/actions/concepts/security/openid-connect), [Best Practices for Managing Secrets in GitHub Actions](https://www.blacksmith.sh/blog/best-practices-for-managing-secrets-in-github-actions)
+**Authority**: GitHub official docs + recognized industry guide
+**Relevant to**: Strategic option for eliminating the rotation toil entirely
+
+**Key Information**:
+
+- The 2026 best practice is **OIDC federation**: no long-lived secrets, every workflow run gets a short-lived, tightly-scoped credential, automatically rotated per run.
+- Supported natively by AWS, Azure, GCP, HashiCorp Vault.
+- **Railway does not document OIDC support** for the CLI as of this research. Confirmed via [search](https://docs.railway.com/cli/deploying) — Railway's published path remains long-lived project tokens.
+- Without OIDC, the fallback best practices are: 30–90 day rotation cadence with calendar reminders, environment-based access controls, and external secret managers (HashiCorp Vault, Infisical) for centralized rotation.
+
+---
+
+### 7. Why "no expiration" guidance in the existing runbook may be unreliable
+
+**Source**: Cross-reference of [Railway Public API docs](https://docs.railway.com/integrations/api) and [Login & Tokens docs](https://docs.railway.com/integrations/oauth/login-and-tokens) and the existing `docs/RAILWAY_TOKEN_ROTATION_742.md`
+**Relevant to**: Why prior rotations have not stuck
+
+**Key Information**:
+
+- The existing runbook (`RAILWAY_TOKEN_ROTATION_742.md`) tells operators: "Expiration: No expiration (critical — do not accept default TTL)".
+- Railway's public docs (as fetched) **do not show a TTL/expiration selector** for account, workspace, or project tokens — only OAuth tokens have documented TTLs.
+- That implies one of three possibilities: (a) the dashboard has a TTL selector that isn't documented; (b) the runbook's premise is incorrect and tokens aren't expiring on a TTL; (c) Railway invalidates tokens for other reasons (suspected leakage, rate limit, account state).
+- The 38-recurrence pattern is inconsistent with a stable TTL. If the TTL were 7 or 30 days the cadence would be regular; the actual cadence (multiple pickups within a few days for the same issue) suggests the token is being invalidated, not naturally expiring.
+
+---
+
+## Code Examples
+
+### Recommended deploy workflow shape per Railway's blog
+
+```yaml
+# From Railway Blog — Using GitHub Actions
+# https://blog.railway.com/p/github-actions
+name: Deploy
+on:
+  push:
+    branches: [main]
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    container: ghcr.io/railwayapp/cli:latest
+    env:
+      RAILWAY_TOKEN: ${{ secrets.RAILWAY_TOKEN }}
+      SVC_ID: <service-id>
+    steps:
+      - uses: actions/checkout@v4
+      - run: railway up --service=${SVC_ID}
+```
+
+Note: this uses the CLI directly (which knows to use the `Project-Access-Token` header) rather than calling the GraphQL `me{id}` query against the project token, which would not work.
+
+---
+
+## Gaps and Conflicts
+
+- **Railway does not publicly document the TTL/expiration options for project, account, or workspace tokens.** The "No expiration" instruction in the existing runbook is not corroborated by the public docs and may be stale or incorrect.
+- **Railway does not document OIDC federation** for CLI auth. If supported, it is undocumented; treat as unavailable until confirmed via Railway support.
+- **No public information on why tokens are being invalidated** server-side beyond expiration. Possibilities (unverified): account ownership change, workspace permission revocation, suspected leakage detection, billing state. The blog post on Railway's [January 2026 incident](https://blog.railway.com/p/incident-report-january-26-2026) was returned by search but doesn't appear directly applicable.
+- **The repo's validation step uses `Authorization: Bearer` + `me{id}` query**, which is incompatible with project tokens. A project token will fail this pre-check even when valid for `railway up`. This is a likely contributor to the recurrence — operators may be rotating to a project token (correct for `railway up`) which then fails the validator (which needs an account/workspace token), and the rotation is judged "broken".
+
+---
+
+## Recommendations
+
+For the human operator who will perform the rotation:
+
+1. **Confirm the token type before pasting.** Per [Railway community](https://station.railway.com/questions/railway-token-invalid-or-expired-59011e20), `RAILWAY_TOKEN` must be a **project token** created at *Project → Settings → Tokens*, not an account/workspace token. Account tokens are silently rejected with the exact "invalid or expired" message we are seeing.
+
+2. **Consider switching to a workspace token + `RAILWAY_API_TOKEN`** if the workflow needs `me{id}`-style account validation. The Railway docs say workspace tokens are the recommended type for "Team CI/CD". This would mean changing both the secret name and the workflow.
+
+3. **Reconsider the validation step.** The current pre-check (`curl … -H "Authorization: Bearer" … me{id}`) is incompatible with project tokens. Either:
+   - Replace the validation with a project-scoped query using the `Project-Access-Token: <token>` header, or
+   - Skip the pre-check and let `railway up` itself fail loud, or
+   - Use a workspace token and `Authorization: Bearer` consistently. (Out-of-scope for this issue per Polecat discipline — should be a separate bead/issue.)
+
+4. **Do not trust "No expiration" as a permanent fix.** The 38-occurrence pattern indicates something else is invalidating the token. After rotation, watch the logs of the next failure: capture the exact `errors[0].message` from the GraphQL response (`Not Authorized` vs. `Token expired` vs. `Project access denied` are different signals).
+
+5. **Long-term: file an enhancement issue** to investigate workspace tokens or external secret-manager integration. Per CLAUDE.md, send mail to mayor for any out-of-scope finding rather than fixing it within the current bead.
+
+For this agent's scope (issue #850):
+
+- Per CLAUDE.md, **do not create a `.github/RAILWAY_TOKEN_ROTATION_*.md` claiming success**.
+- The investigation document (already produced for prior occurrences) should reference `docs/RAILWAY_TOKEN_ROTATION_742.md` and add the new findings above as updated context.
+
+---
+
+## Sources
+
+| # | Source | URL | Relevance |
+|---|--------|-----|-----------|
+| 1 | Railway Public API Docs | https://docs.railway.com/integrations/api | Token types, scopes, headers |
+| 2 | Railway CLI Guide | https://docs.railway.com/guides/cli | `RAILWAY_TOKEN` vs `RAILWAY_API_TOKEN` precedence |
+| 3 | Railway Blog — GitHub Actions | https://blog.railway.com/p/github-actions | Official deploy pattern using project tokens |
+| 4 | Railway Help Station — RAILWAY_TOKEN invalid or expired | https://station.railway.com/questions/railway-token-invalid-or-expired-59011e20 | Project-token-only acceptance for `RAILWAY_TOKEN` |
+| 5 | Railway Help Station — CLI Unauthorized with RAILWAY_TOKEN | https://station.railway.com/questions/cli-throwing-unauthorized-with-railway-24883ba1 | Alternative failure modes |
+| 6 | Railway Help Station — Token for GitHub Action | https://station.railway.com/questions/token-for-git-hub-action-53342720 | Workspace vs account token nuance |
+| 7 | Railway Login & Tokens Docs | https://docs.railway.com/integrations/oauth/login-and-tokens | OAuth TTL (1h access, 1yr refresh) |
+| 8 | Railway Deploying with the CLI | https://docs.railway.com/cli/deploying | Project token usage with `railway up` |
+| 9 | GitHub OpenID Connect Docs | https://docs.github.com/en/actions/concepts/security/openid-connect | OIDC alternative to long-lived tokens |
+| 10 | Best Practices for Managing Secrets in GitHub Actions | https://www.blacksmith.sh/blog/best-practices-for-managing-secrets-in-github-actions | 30–90 day rotation cadence, external secret managers |
+| 11 | Railway Incident Report Jan 2026 | https://blog.railway.com/p/incident-report-january-26-2026 | Returned by search; no direct token-invalidation link |

